ansible_collections/arista/avd/roles/eos_designs/templates/overlay/ibgp/address-family-vpn-ipv6.j2

```diff
@@ -1,14 +1,23 @@
-{% if switch.mpls_overlay_role in ["server", "client"] %}
+{% if switch.mpls_overlay_role in ["server", "client"] or (switch.mpls_vpn_gateway is arista.avd.defined(true) and "vpn-ipv6" in switch.mpls_vpn_gateway_address_families) %}
```

Comment:
You are adding special checks everywhere for the gateway feature. It seems like you should just set the mpls_overlay_role to something meaningful (ex. client). I guess a gateway could also work as client/server ?

Reply:
If i reuse the mpls_overlay_role, it would break ibgp design due to these lines in the ibgp evpn template:

```jinja2
  address_family_evpn:
{% if switch.mpls_overlay_role in ["server", "client"] %}
    neighbor_default:
      encapsulation: mpls
{%     if switch.mpls_overlay_role == "client" %}
      next_hop_self_source_interface: Loopback0
{%     endif %}
{% endif %}
```

I suppose i could test for the inverse here: if server/client and not mpls_vpn_gateway.enabled. There are many more instances where this would be required in the ibgp overlay templates. Not sure which is better, adding checks for gateway knobs where i need mpls functionality turned on or adding checks for gateway knobs where i need to prevent mpls functionality from getting turned on as a result of mpls_overlay_role reuse...